### PR TITLE
[READY] add all entries overview and CSV download

### PR DESCRIPTION
**Pull request by jurre**

### CODE REVIEW

**Comment by jurre** [comment]
Could use some thoughts on the design, this breaks spectacularly on mobile:

screen shot 2015-01-06 at 09 46 56

**Comment by dkhgh** [comment]
I started exploring a more card-like approach. Wireframe is on another PR or issue. To be continued

**Comment by jurre** [comment]
@dkhgh I love the card-like approach but was thinking that since the reason that we're even displaying these entries here is so that you can *download* them as CSV so I wanted to make it look as much like the data that you'll download.

But maybe it would be nice to just show the same cards everywhere, especially once we add filtering etc to this view.

**Comment by dkhgh** [comment]
Agreed. 
But getting a csv export on mobile is probably not gonna happen too often so we shouldn't worry too much about that view

**Comment by jurre** [comment]
Yeah, word.

File: app/controllers/entries_controller.rb [hunk 1]
```diff
@@ -12,10 +12,17 @@ def create
...
   def index
-    @user = User.find_by_slug(params[:user_id])
+    if scoped?
+      @user = User.find_by_slug(params[:user_id])
+      @entries = @user.entries.by_date.page(params[:page]).per(20)
+    else
+      @entries = Entry.by_date.page(params[:page]).per(20)
+      render template: "entries/report" unless params[:format] == "csv"
+    end
+
```
Comment (Marthyn) [inline app/controllers/entries_controller.rb]: EntriesReportsController right? i'd say yes.

File: app/models/report.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ def category
   end
 
   def client
-    @entry.client.try(:name)
+    @entry.project.client.try(:name)
```
Comment (Marthyn) [inline app/models/report.rb]: why not @entry.client? is has a has_one association

File: app/views/entries/report.html.haml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      %tr.info-row
+        %td= entry.date
+        %td= link_to entry.user.full_name, entry.user
+        %td= link_to entry.project.name, entry.project
+        %td= link_to entry.category.name, entry.category
+        %td= link_to entry.project.client.name, entry.project.client if entry.project.client
+        %td= entry.hours
+        %td= entry.project.billable
+        %td= sanitize autolink_tags(entry.description)
+  = paginate @entries
```
Comment (Marthyn) [inline app/views/entries/report.html.haml]: Maybe add a billed column? Or is that only in my PR ? :P

Comment (jurre) [inline app/views/entries/report.html.haml]: Yep, it is

File: app/models/report.rb (hunk 2 above)
Comment (jurre) [inline app/models/report.rb]: Only in your current PR :)

**Comment by jurre** [comment]
Made them filterable, but not sure what to do with all this data on the screen:

screen shot 2015-01-11 at 19 11 14

**Comment by jurre** [comment]
Also, we'll probably want some other/more filters here, thoughts?

**Comment by Marthyn** [comment]
Per category seems interesting.

**Comment by Marthyn** [comment]
It would also be nice if we see a total amount of hours. That said, would be nice if we have that everywhere throughout the overviews.

**Comment by jurre** [comment]
Yep, agreed. Holding off with adding filters till we get some feedback from the business people though, so LGTM :boom:

Looks like this now:
screen shot 2015-01-12 at 10 59 58

**Comment by jurre** [comment]
@DefactoSoftware/hours can we merge this?

**Comment by Marthyn** [comment]
Yes

**Pull request merged**